container-host-files/opt/scf/bin/klog.sh

```diff
@@ -5,192 +5,82 @@ set -e
 KLOG=${HOME}/klog
 
 if [ "$1" == "-h" ]; then
-    cat <<EOF
+	cat <<EOF
 usage: $0 [-f] [-v] [INSTANCE_ID]
 
   -f  forces fetching of all logs even if a cache already exists
 
   INSTANCE_ID defaults to "scf"
 EOF
-    exit
+	exit
 fi
 
 FORCE=0
 if [ "$1" == "-f" ]; then
-    shift
-    FORCE=1
+	shift
+	FORCE=1
 fi
 
 NS=${1-scf}
 DONE="${KLOG}/${NS}/done"
 
-if [ "${FORCE}" == "1" ] ; then
-    rm -f "${DONE}" 2> /dev/null
+if [ "${FORCE}" == "1" ]; then
+	rm -f "${DONE}" 2>/dev/null
 fi
 
 function get_phase() {
-  kubectl get pod "${POD}" --namespace "${NS}" --output=jsonpath='{.status.phase}'
+	kubectl get pod "${POD}" -n "${NS}" -o jsonpath='{.status.phase}'
 }
 
 function check_for_log_dir() {
-  kubectl exec "${POD}" --namespace "${NS}" -- bash -c "[ -d /var/vcap/sys/log ]" 2> /dev/null
+	kubectl exec "${POD}" -n "${NS}" -- bash -c "[ -d /var/vcap/sys/log ]" 2>/dev/null
 }
 
 if [ ! -f "${DONE}" ]; then
-    rm -rf "${KLOG:?}/${NS:?}"
-    mkdir -p "${KLOG}/${NS}"
+	rm -rf "${KLOG:?}/${NS:?}"
+	BASE_DIR="${KLOG}/${NS}"
+	mkdir -p ${BASE_DIR}
 
-    PODS=($(kubectl get pods --namespace "${NS}" --output name --show-all=true | sed 's/pods\?\///'))
+	# Retrieving all the pods
+	PODS=($(kubectl get pods -n cf -o jsonpath='{.items[*].metadata.name}'))
 
-    for POD in "${PODS[@]}"; do
-        DIR="${KLOG}/${NS}/${POD}"
+	for POD in "${PODS[@]}"; do
 
-        mkdir -p "${DIR}"
+		echo "Dumping logs for pod: ${POD}"
 
-        # Get the CF logs inside the pod if there are any
-        if [ "$(get_phase)" != 'Succeeded' ] && check_for_log_dir; then
-            kubectl cp --namespace "${NS}" "${POD}":/var/vcap/sys/log/ "${DIR}/" 2> /dev/null
-        fi
+		POD_DIR="${BASE_DIR}/${POD}"
+		mkdir -p ${POD_DIR}
 
-        # Get the pod logs - previous may not be there if it was successful on the first run.
-        # Unfortunately we can't get anything past the previous one
-        kubectl logs "${POD}" --namespace "${NS}" > "${DIR}/kube.log"
-        kubectl logs "${POD}" --namespace "${NS}" --previous > "${DIR}/kube-previous.log" 2> /dev/null || true
-        kubectl describe pods "${POD}" --namespace "${NS}" > "${DIR}/describe-pod.txt"
-    done
+		# Get the CF logs inside the pod if there are any
+		if [ "$(get_phase)" != 'Succeeded' ]; then
+			if [ check_for_log_dir ]; then
+				kubectl cp -n "${NS}" "${POD}":/var/vcap/sys/log/ "${POD_DIR}/" 2>/dev/null
```

Comment:
`set -e` (line 3 of the script) should still be active; so the old `|| true` was to temporarily disable that behaviour.

Is that now fine to have?

Also, should this also be done per-container? Multiple containers (probably) have different `/var/vcap/sys/log/` directories. (When we were talking offline, I had thought you merged it all, instead of only grabbing it from one container)

Reply:
For #1 as discussed, we will keep it as such.

For #2, I have made changes to include `/var/vcap/sys/log/` logs from containers.